WebApi/WebApi/Library/LibraryManager.cs

```
@@ -0,0 +1,237 @@
+            this.books.Add(new Book(4, "Pride and Prejudice", 3, 1813));
+            this.books.Add(new Book(5, "Fahrenheit 451", 4, 1953));
+            this.books.Add(new Book(6, "Dandelion Wine", 4, 1957));
+            this.books.Add(new Book(7, "The Lovely Bones", 5, 2002));
+        }
+
+        /// <summary>
+        /// Add new author
+        /// </summary>
+        /// <param name="author">author's instance</param>
+        /// <returns>the result of an action</returns>
+        public Author AddAuthor(Author author)
+        {
+            this.authors.Add(author);
+            return author;
+        }
+
+        /// <summary>
+        /// Add new book
+        /// </summary>
+        /// <param name="book">book's instance</param>
+        /// <returns>the result of an action</returns>
+        public Book AddBook(Book book)
+        {
+            this.books.Add(book);
+            return book;
+        }
+
+        /// <summary>
+        /// Get all books
+        /// </summary>
+        /// <returns>book collection</returns>
+        public IEnumerable<Book> GetBooks()
+        {
+            return this.books;
+        }
+
+        /// <summary>
+        /// Get all books
+        /// </summary>
+        /// <returns>book collection</returns>
+        public IEnumerable<Author> GetAuthors()
+        {
+            return this.authors;
+        }
+
+        /// <summary>
+        /// Get book by id
+        /// </summary>
+        /// <param name="id">book's id</param>
+        /// <returns>current book</returns>
+        public Book GetBook(int id)
+        {
+            Book currentBook = null;
+            foreach (var book in this.books)
+            {
+                if (book.Id == id)
+                {
+                    currentBook = book;
+                }
+                else
+                {
+                    throw new ArgumentException("No book with this id");
```

You are throwing an exception for each item in the list of books if id of the book is not what you are looking for. Have you tested this method, I don't think that it working at all! 

You should throw an exception only in case there was no book with such id in the entire collection!